pkg/controller/perconaservermongodbbackup/perconaservermongodbbackup_controller.go

```diff
@@ -228,14 +229,28 @@ func (r *ReconcilePerconaServerMongoDBBackup) reconcile(
 	}
 
 	if cjobs {
-		if cr.Status.State != psmdbv1.BackupStateWaiting {
-			log.Info("Waiting to finish another backup/restore.")
-		}
+		log.Info("Waiting to finish another backup/restore.")
 		status.State = psmdbv1.BackupStateWaiting
 		return status, nil
 	}
 
-	switch cr.Status.State {
+	log.Info("Acquiring the backup lock")
+	lease, err := k8s.AcquireLease(ctx, r.client, naming.BackupLeaseName(cluster.Name), cr.Namespace, naming.BackupHolderId(cr))
+	if err != nil {
+		return status, errors.Wrap(err, "acquire backup lock")
+	}
+
+	if lease.Spec.HolderIdentity != nil && *lease.Spec.HolderIdentity != naming.BackupHolderId(cr) {
+		log.Info("Another backup is holding the lock", "holder", *lease.Spec.HolderIdentity)
+		status.State = psmdbv1.BackupStateWaiting
+		return status, nil
+	}
```

Comment:
If the backup has finished or failed, will the operator still try to acquire the lease?

Reply:
no, this code is not executed for finished or failed backups: https://github.com/percona/percona-server-mongodb-operator/blob/b4aeb8f7cd96a0e4e9ae94edee0004390b141074/pkg/controller/perconaservermongodbbackup/perconaservermongodbbackup_controller.go#L128-L131